tracardi/process_engine/action/v1/converters/base64/encode/plugin.py

```diff
@@ -0,0 +1,78 @@
+from base64 import b64encode
+
+from tracardi.service.plugin.domain.register import Documentation, Form, FormComponent, FormField, FormGroup, \
+    MetaData, Plugin, PortDoc, Spec
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.runner import ActionRunner
+from .model.models import Configuration
+
+
+def validate(config: dict):
+    return Configuration(**config)
+
+
+class Base64EncodeAction(ActionRunner):
+    config: Configuration
+
+    async def set_up(self, init):
+        self.config = validate(init)
+
+    async def run(self, payload: dict, in_edge=None):
+        dot = self._get_dot_accessor(payload)
+        text = dot[self.config.to_base64]
+        result = b64encode(text.encode(self.config.input_encoding)).decode('ascii')
+
+        return Result(port="payload", value={"base64": result})
```

Comment:
This hardcodes the key in the result. Does it make sense to reuse the key from the input? Is this possible? Using the same key would allow "transparent" actions, that can be inserted between two existing actions without affecting the result.

Or perhaps apply base64 to multiple keys in the input payload? (If possible).

I'm also happy to implement this in a follow-up PR.

Reply:
Will leave for now. If required later, I could introduce a config option `sink` which lets users override the default output key.

Reply:
OK